### ISSUE TITLE
Shouldn't be able to resize a pane when there is only one

### ISSUE THREAD

**Issue by lee-dohm** [atom/atom#11673]
### Prerequisites
- [x] Can you reproduce the problem in [safe mode](http://flight-manual.atom.io/hacking-atom/sections/debugging/#check-if-the-problem-shows-up-in-safe-mode)?
- [x] Are you running the [latest version of Atom](http://flight-manual.atom.io/hacking-atom/sections/debugging/#update-to-the-latest-version)?
- [x] Did you check the [debugging guide](http://flight-manual.atom.io/hacking-atom/sections/debugging/)?
- [x] Did you check the [FAQs on Discuss](https://discuss.atom.io/c/faq)?
- [x] Are you reporting to the [correct repository](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#atom-and-packages)?
- [x] Did you [perform a cursory search](https://github.com/issues?q=is%3Aissue+user%3Aatom+-repo%3Aatom%2Felectron) to see if your bug or enhancement is already reported?
### Description

Atom gets into a perplexing state for some people ([for example](https://discuss.atom.io/t/atom-editor-width-bug/28439)) whether by some combination of keys or package actions where there is only one pane, but it does not take up the full width of the editor window. Atom should not be able to get into this state.
### Steps to Reproduce
1. Launch Atom
2. Ensure there is only one pane in the window
3. Press <kbd>Cmd+Alt+-</kbd> multiple times

**Expected behavior:** Nothing to change

**Actual behavior:** The pane area for the single pane shrinks even though the window does not change size
### Versions

Atom    : 1.9.0-dev-5a9bd28
Electron: 0.36.8
Chrome  : 47.0.2526.110
Node    : 5.1.1

apm  1.9.3
npm  2.13.3
node 0.10.40
python 2.7.10
git 2.8.1

**Comment by lee-dohm** [comment]
Atom also shouldn't ever restore a resized single pane when starting up, so people shouldn't have to launch with `atom --clear-window-state` to fix this if they get into the weird state.

**Comment by moehuh** [comment]
Experienced exactly this problem. `atom --clear-window-state` solves it, but I agree: Atom should not restore a resized single pane!

**Comment by randohinn** [comment]
Experienced this. `atom --clear-window-state`was a fix. I suggest adding a menu item for it to the View menu.

**Comment by lee-dohm** [comment]
@randohinn I will strongly advocate against adding a Clear Window State item to any menu because it blows away all stored state, including any unsaved changes. Also, clearing the window state shouldn't be something that we encourage people to do ... it should be a workaround, not a common practice.

**Comment by DamnedScholar** [comment]
It could be buried under `View -> Developer` without a keyboard shortcut. Many people don't use the command line, so having to pop open a console in order to reopen Atom in a special mode is extra work.

**Comment by lee-dohm** [comment]
> It could be buried under View -> Developer without a keyboard shortcut.

None of the other items under View &rarr; Developer will cause people to lose unsaved work. To be clear, Clear Window State will not only lose unsaved work for the current project but **all** projects. We don't want people to be casually using this, so the pain of popping open a console and reopening Atom is intentional.

**Comment by Bjorn-c** [comment]
Took me an **hour** before I found this solution, +1 for adding a menu item for this.

**Comment by DamnedScholar** [comment]
@Bjorn-c Where did you look? With [the Debugging section of the Flight Manual](http://flight-manual.atom.io/hacking-atom/sections/debugging/) and [a FAQ entry on the forum](https://discuss.atom.io/t/im-having-a-problem-with-atom-what-are-the-standard-debug-steps/19435) that refers to the Flight Manual, it's pretty easy to get to that information. Did you read that and dismiss it as probably not relevant to your situation? Did you not look there?

**Comment by Bjorn-c** [comment]
I indeed dismissed it ...

**Comment by DamnedScholar** [comment]
Good to know. Maybe that needs to be made more prominent.

**Comment by ibqn** [comment]
I have just installed the latest stable atom version on windows 10 and ran into this empty space bug issue:/

after running `$ atom --clear-window-state` this glitch went away...

**Comment by lorenzos** [comment]
Any news on this? :disappointed:

**Comment by lee-dohm** [comment]
Not from the maintainer team, no. If someone wanted to help fix this, I'd be very happy to review a well-written pull request :+1:

**Comment by 8-uh** [comment]
Always the outlier, `--clear-window-state` did not work for me...

But `cmd-alt-+` did.

**Comment by richcocoa** [comment]
The relevant bit of code seems to be in `src/pane-element.coffee` around line `123`

```coffee
  flexScaleChanged: (flexScale) ->
    @style.flexGrow = flexScale
```

What is needed is probably a check to see the number of items in the pane of the pane element before changing the `flex-grow` style. Should we also check if it has a text editor element inside it?

I would have submitted the PR, but I couldn't find a way to get the pane of the current pane element. Can someone who is familiar with the code base help me?

**Comment by lee-dohm** [comment]
> Should we also check if it has a text editor element inside it?

No, because it could be the Settings View and still have the same problem:

<img width="1353" alt="screen shot 2017-05-25 at 4 23 22 pm" src="https://cloud.githubusercontent.com/assets/1038121/26474547/81238626-4166-11e7-90de-cb222b09d781.png">

>  I couldn't find a way to get the pane of the current pane element. 

If you want the pane of the active pane item (what you're calling pane element), you can use [`atom.workspace.getActivePane()`](https://atom.io/docs/api/latest/Workspace#instance-getActivePane). This is because the active pane item will always be in the active pane.

**Comment by richcocoa** [comment]
@lee-dohm Could you review my PR #14665?

**Comment by ernesto28** [comment]
Is this fixed? I want to help, I'm new in open source contributions

I notice that it can happen even with more than one pane when they are in a horizontal row and you resize all of them

![image](https://user-images.githubusercontent.com/32112295/34175347-6cc98710-e4b9-11e7-919d-a2908a4a90ff.png)

But when you close one of the open panes the other panes get back to normal

![image](https://user-images.githubusercontent.com/32112295/34175365-79ac2ad2-e4b9-11e7-972c-26c375eeadb8.png)

This are the versions I have:
![image](https://user-images.githubusercontent.com/32112295/34175478-c4e82014-e4b9-11e7-94ec-e3008b4f58e0.png)

**Comment by lee-dohm** [comment]
No, this isn't fixed. You can use the [debugging guide](http://flight-manual.atom.io/hacking-atom/sections/debugging/) in the [Atom Flight Manual](http://flight-manual.atom.io) to determine if you're on the latest version. (You should start by reproducing the problem in the latest stable version and then start development from the latest development version.)

**Comment by ernesto28** [comment]
I make a PR fixing it for one pane but I'm still looking how to solve it with more than one.
@lee-dohm is there a way to compare the width of the container with the width of the panes that are inside him?

**Comment by CharlotteDunois** [comment]
Any updates on this? Having to clear the window state all the time isn't a valid fix for me.

**Comment by Ben3eeE** [comment]
@lee-dohm Do you think this can be closed now that #16428 is merged? I don't think https://github.com/atom/atom/issues/11673#issuecomment-216374261 has been addressed.

**Comment by lee-dohm** [comment]
Yeah, this can be closed. The deserializing thing has a workaround in `--clear-window-state` so while it's less-than-ideal, it is a well-known feature at this point.

Thanks for the reminder @Ben3eeE :bow:
